### PR TITLE
use score in PathLocation to remove the need for Candidate. fixes #42

### LINKED ISSUE #42: Remove candidate.h
There seems no special reason to use the `Candidate` as a wrapper of `baldr::PathLocation`. They are the same thing; we should use `PathLocation` directly.

### PR DESCRIPTION
**Pull request by kevinkreiser**
still need to fetch and merge changes from @ptpt last pr

### CODE REVIEW

**Comment by kevinkreiser** [comment]
@ptpt can you have a look when you get a chance

File: src/meili/map_matcher.cc [hunk 1]
```diff
@@ -259,7 +259,7 @@ FindMatchResult(const MapMatching::state_iterator& previous_state,
   const auto& c = state.candidate();
   //Note: technically a candidate can have correlated to more than one place in the graph
   //but the way its used in meili we only correlated it to one place so .front() is safe
-  return {c.edges.front().projected, c.distance(), edgeid, state.id()};
+  return {c.edges.front().projected, c.edges.front().score, edgeid, state.id()};
```
Comment (ptpt) [inline src/meili/map_matcher.cc]: Should it be `sqrt(c.edges.front().score)` if the score is the squared distance?

File: valhalla/meili/geojson_writer.h [hunk 2]
```diff
@@ -138,7 +138,7 @@ void serialize_state(rapidjson::Writer<buffer_t>& writer,
   writer.Uint(state.time());
 
   writer.String("distance");
-  writer.Double(state.candidate().distance());
+  writer.Double(state.candidate().edges.front().score);
```
Comment (ptpt) [inline valhalla/meili/geojson_writer.h]: The same I think it should be `sqrt(state.candidate().edges.front().score)` here.

**Comment by kevinkreiser** [comment]
ok ive resolved that. wondering if i should do the sqrt when setting it so it might be more clear. also what is the purpose of using square distance at all? in terms of computing distances in polar coordinates getting the square distance is actually more expensive than getting the not-squared-distance

**Comment by kevinkreiser** [comment]
build will fail until baldr is merged

**Comment by ptpt** [comment]
I'm fine with both usages. The main reason of using distance square is that internally we use distance square more than we use distance. We get distance square from the projection calculation and we apply it directly in the emission cost calculation. Also it seems that the approximator (used in the project function) doesn't offer a cheap distance method yet.

:ship: when it's ready.

**Comment by kevinkreiser** [comment]
ah i see, so basically because it uses the approximator. i did notice that yesterday but it slipped my mind! thanks for the info!

**Pull request merged**